lib/commentOnOpenedPullRequest.js

```diff
@@ -116,11 +117,35 @@ commentOnOpenedPullRequest._askAboutChanges = function (files, baseBranch) {
     return true;
 };
 
-commentOnOpenedPullRequest._askAboutContributors = function (userName, repositorySettings, headApiUrl, headBranch) {
-    if (!defined(repositorySettings.contributorsPath)) {
+commentOnOpenedPullRequest._askAboutContributors = function (userName, repositorySettings, headApiUrl, headBranch, repositoryContributorsUrl) {
+    if (!defined(repositorySettings.contributorsPath) && !repositorySettings.contributorsFromGitHub) {
         return Promise.resolve(false);
     }
 
+    if (repositorySettings.contributorsFromGitHub) {
+        var contributorsPromises = [];
+        for (var i = 1; i < 6; i++) { // GitHub maintains only 500 named contributors
```

Comment:
Is there a reason we are manually requesting each page rather than using [pagination](https://docs.github.com/en/rest/guides/traversing-with-pagination)? This makes 5 requests whether the pages exist or not, which we should avoid.

Reply:
@ggetz That was one of the things I had considered, but decided to take this approach as it has a maximum of 5 requests. My reasoning was that GitHub maintains a list of maximum 500 contributors. If we request all the pages, then we can run a single search, rather than 5 separate searches.

Of course, most projects have less than 100 contributors so we could get away with a paginated version that is pretty much the same in the common use-cases.

Reply:
If we go to a paginated approach, there's no need change to search each page individually (though you could). We can concatenate all the results and search that just as you are doing before. Paginating is about making fewer requests, especially if we expect there to be one page most of the time, and will present less of a chance for one of those requests to fail for whatever reason and throw an error.

Reply:
@ggetz Updated.